lib/diff.js

```diff
@@ -110,16 +110,33 @@ const getAction = ({actual, ideal}) => {
   if (ideal.isRoot && actual.isRoot)
     return null
 
+  // if the versions don't match, it's a change no matter what
+  if (ideal.version !== actual.version)
+    return 'CHANGE'
+
   const binsExist = ideal.binPaths.every((path) => existsSync(path))
 
   // top nodes, links, and git deps won't have integrity, but do have resolved
-  if (!ideal.integrity && !actual.integrity && ideal.resolved === actual.resolved && binsExist)
+  // However! if *neither* has a resolved value, then we need to refetch
+  // to figure out what it even is, so we mark as CHANGE.
+  const noIntegrity = !ideal.integrity && !actual.integrity
+  const resolvedMatch = ideal.resolved && ideal.resolved === actual.resolved
+  if (noIntegrity && resolvedMatch && binsExist)
+    return null
+
+  // If NEITHER has resolved OR integrity, but the versions match
+  // (which we know because of the check above), and we don't have
+  // to find bins, then we should just leave it as-is.
+  const noResolved = !ideal.resolved && !actual.resolved
+  if (noIntegrity && noResolved && binsExist)
```

Comment:
There are to nearly identical if statements here, that differ only by `resolvedMatch` or `noResolved`.  This may be one of the rare times I advocate for combining the two, and updating the comment to reflect the single conditional.